tests/context/server.rs

```diff
@@ -22,19 +18,18 @@ pub struct EchoServer {
 pub enum Error {}
 
 impl EchoServer {
-    pub async fn start() -> Self {
+    pub async fn start(config_from_env: &Config) -> Self {
         let public_port = get_random_port();
-        let config: Config = Config {
+        let config = Config {
             port: public_port,
             public_url: format!("http://127.0.0.1:{public_port}"),
             log_level: "info,echo-server=info".into(),
             log_level_otel: "info,echo-server=trace".into(),
             disable_header: true,
-            relay_url: "https://relay.walletconnect.com".into(),
+            relay_public_key: config_from_env.relay_public_key.clone(),
             validate_signatures: false,
-            database_url: DATABASE_URL.into(),
-            #[cfg(feature = "multitenant")]
-            tenant_database_url: TENANT_DATABASE_URL.into(),
+            database_url: config_from_env.database_url.clone(),
+            tenant_database_url: config_from_env.tenant_database_url.clone(),
```

Comment:
Looks like we lost the `#[cfg(feature = "multitenant")]`

```suggestion
            #[cfg(feature = "multitenant")]
            tenant_database_url: config_from_env.tenant_database_url.clone(),
```

Reply:
Thanks for catching it, but It was not added because of `TENANT_DATABASE_URL` [was hardcoded ](https://github.com/WalletConnect/echo-server/blob/591effe09eb1ca8e0b63011f785a06c0b4ca9c86/tests/context/mod.rs#L14) before and to use it conditionally more refactoring is needed (like we opening both connections for a pg pool, etc.). 
As this PR became pretty big for a review, this refactoring should go as a follow-up.